components/ReportLadderMatch.js

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react';
+import Select from 'react-select';
+import { getUpdatedRatings } from '../lib/elo';
+import { trimDecimals } from '../lib/helpers';
+import Loading from './Loading';
+
+const ReportLadderMatch = ({ players, onReportedMatch, loading }) => {
+  const [challenger, setChallenger] = useState(null);
+  const [opponent, setOpponent] = useState(null);
+  const [challengerOptions, setChallengerOptions] = useState([]);
+  const [opponentOptions, setOpponentOptions] = useState([]);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    setError('');
+  }, [challenger, opponent]);
+
+  useEffect(() => {
+    setChallengerOptions(
+      [...players]
+        .sort((a, b) => a.ladderPosition - b.ladderPosition)
+        .filter((player) => (player.ladderPosition != 1))
+        .map((player) => ({
+          value: player._id,
+          label: `(${player.ladderPosition}) ${player.name} `,
+        })),
+    );
+  }, [players]);
+
+  useEffect(() => {
+    if (!challenger || players.length < 0) {
+      return
+    }
+    const challengerPosition = players.find((player) => player._id === challenger.value).ladderPosition
+    setOpponentOptions(
+      [...players]
+        .sort((a, b) => a.ladderPosition - b.ladderPosition)
+        .filter((player) => player.ladderPosition < challengerPosition && player.ladderPosition > challengerPosition - 3 )
```

Comment:
For readability reasons, could we maybe place the magic number "3" in a variable?